Yoast/wordpress-seo
# Move Ryte integration to site health

## Summary

* Introduces a new Site Health Check for Ryte. 
* Removes Ryte Integration on the dashboard.
* Removes Ryte `wpseo-dismiss-onpageorg` notification.
* Refactors all `onpage` to `ryte`.
* Adds test class for Ryte Site Health Check.
* Introduces new upgrade routine `xx.x`.


This PR can be summarized in the following changelog entry:

* Moved the Ryte notice and widget from the dashboard to WordPress' Health Check

## Relevant technical choices:

* For the deprecations and upgrade routine a temporary `xx.x` for the plugin version was used. This should be updated when merging into trunk once the actual milestone will be determined.

## Test instructions

This PR can be tested by following these steps:
* Checkout this branch.
* Go to the `Dashboard`, there should be no `Indexability check by Ryte` in the Yoast widget. 
* Go to `Tools` > `Site Health` and see an Ryte health check under `recommended improvements`.

__Ryte not enabled__
* Go to `SEO` >`General` > `Features` > choose `"Off"` for the `Ryte integration` toggle and save your changes.
* Go to `Site Health`, there should be no response for Ryte (also not under `Passes tests`). 
* Switch the `Ryte integration` toggle back to `"On"` and save your changes.
* Go to `Site Health`, the Ryte response should have reappeared.

__Site not public__
* Go to `Settings` > `Reading` > tick the `Search Engine Visibility` box to discourage search engines from indexing the site and save your changes. 
* Go to `Site Health`, there should be no response for Ryte (also not under `Passes tests`). 
* A notification should appear in `SEO` > `General` starting with `"Huge SEO issue" ... `.
* Go to `Settings` > `Reading` > untick the `Search Engine Visibility` box and save your changes. 
* Go to `Site Health`, the Ryte response should have reappeared.

__Can't tell if it can be indexed__
* Go to `Site Health`, under `recommended improvements` should be an SEO response as seen in the image below. 
* Test whether the links and the button work. 
<img width="700" alt="unknown_indexability" src="https://user-images.githubusercontent.com/43582255/71621424-763aa180-2bcf-11ea-9055-d7a63c16ab5d.png">

__Can be indexed__
* In `class-ryte-request.php`  add  `$target_url = "https://yoast.com";` on the first line of the function `get_remote`.
* Go to `Site Health`, under `Passed tests` should be an SEO response as shown in the image below. 
<img width="700" alt="is_indexable" src="https://user-images.githubusercontent.com/43582255/71621708-3d033100-2bd1-11ea-9a2e-8e1090fa4499.png">

__Cannot be indexed__
* In `health-check-ryte.php` change `switch ( $ryte_option->get_status() )` to `switch ( 0 )` (_there is probably another / better way to test this, but I'm not sure how..._).
* Go to `Site Health`, under `critical issues` should be an SEO response as seen in the image below. 
* Test whether the links and the button work. 
<img width="700" alt="is_not_indexable" src="https://user-images.githubusercontent.com/43582255/71621756-78056480-2bd1-11ea-8052-701d173ba4a6.png">

__Upgrade routine__
Trigger the upgrade routine:
* Check whether the `wpseo-dismiss-onpageorg` notification is correctly removed from the database.
* Check whether `onpage_indexability` is changed to `ryte_indexability` (contained in the `option_value` of `wpseo` in the `wp-options` table).
* Check whether  `wpseo_onpage` is changed to ~~ryte_onpage~~ `wpseo_ryte` [Edit by Andrea] (`option_name` in the `wp_options` table).


## UI changes

## Documentation

## Quality assurance


Fixes #12897
